airflow/utils/cli.py

```diff
@@ -102,8 +102,9 @@ def wrapper(*args, **kwargs):
                 # Check and run migrations if necessary
                 if check_db:
                     from airflow.utils.db import check_and_run_migrations, synchronize_log_template
-
-                    check_and_run_migrations()
+                    from airflow.configuration import conf
+                    if conf.get("database", "CHECK_MIGRATIONS"):
```

Comment:
```suggestion

                    if conf.getboolean("database", "check_migrations"):
```

(It is sort of a pattern of having the config name in uppercase, but one I'm trying to slowly remove)